amgcl/detail/qr.hpp

```diff
@@ -223,10 +223,10 @@ class QR {
             const int row_stride = (Order == row_major ? nmax : 1);
             const int col_stride = (Order == row_major ? 1 : m);
 
-            for(int i = 0, ii = 0; i < n; ++i, ii += row_stride + col_stride)
+            for(int i = 0, ii = 0; i < m; ++i, ii += row_stride + col_stride)
```

Comment:
In other words, I am not sure if we can correctly  solve an underdetermined system here. What do you think?